src/miniscript/analyzable.rs

```diff
@@ -22,7 +22,115 @@ use core::fmt;
 use std::error;
 
 use crate::prelude::*;
-use crate::{Miniscript, MiniscriptKey, ScriptContext};
+use crate::{Miniscript, MiniscriptKey, ScriptContext, Terminal};
+
+/// Params for parsing miniscripts that either non-sane or non-specified(experimental) in the spec.
+/// Used as a parameter [`Miniscript::from_str_insane`] and [`Miniscript::parse_insane`].
+///
+/// This allows parsing miniscripts if
+/// 1. It is unsafe(does not require a digital signature to spend it)
+/// 2. It contains a unspendable path because of either
+///     a. Resource limitations
+///     b. Timelock Mixing
+/// 3. The script is malleable and thereby some of satisfaction weight
+///    guarantees are not satisfied.
+/// 4. It has repeated public keys
+/// 5. raw pkh fragments without the pk. This could be obtained when parsing miniscript from script
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Copy, Default)]
+pub struct ExtParams {
+    /// Allow parsing of non-safe miniscripts
+    pub top_unsafe: bool,
+    /// Allow parsing of miniscripts with unspendable paths
+    pub resource_limitations: bool,
+    /// Allow parsing of miniscripts with timelock mixing
+    pub timelock_mixing: bool,
+    /// Allow parsing of malleable miniscripts
+    pub malleability: bool,
+    /// Allow parsing of miniscripts with repeated public keys
+    pub repeated_pk: bool,
+    /// Allow parsing of miniscripts with raw pkh fragments without the pk.
+    /// This could be obtained when parsing miniscript from script
+    pub raw_pkh: bool,
+}
```

Comment:
I really love this struct. It's an easy reference place for basically everything weird about Miniscript :)